### ISSUE TITLE
PS1 Redump DAT Processing Issue

### ISSUE THREAD

**Issue by JbstormburstADV** [unexpectedpanda/retool#288]
**Describe the bug**
During processing of DAT file, ReTool outputs during step 2 of processing "Retool has had an unexpected error."  Attaching trace below and attaching DAT file as text file to submission.
```
Traceback (most recent call last):
  File "retoolgui.py", line 175, in run
  File "retool.py", line 215, in main
  File "modules\chooseparent.py", line 1777, in choose_parent
  File "modules\chooseparent.py", line 2179, in choose_parent_process
KeyError: <modules.dats.DatNode object at 0x000001B817103C50>

The error occurred on this file:
D:\DAT_Files\Sony - PlayStation - Datfile (10747) (2023-10-03 23-04-12).dat
```
**Operating system**
Windows 11

**Retool edition**
retool-gui.exe

**Retool version**
v2.01.5

**Are you using custom global or system filters? If so, list them below**
N/A

**What other settings are you using?**
- Exclude bad dumps, manuals, demos/kiosks/samples, educational, MIA
- Prefer regions over languages
- Use original input DAT header
[Sony - PlayStation - Datfile (10747) (2023-10-03 23-04-12).txt](https://github.com/unexpectedpanda/retool/files/12799329/Sony.-.PlayStation.-.Datfile.10747.2023-10-03.23-04-12.txt)

**Comment by unexpectedpanda** [comment]
Thanks for the report!

Usually something like this is caused by an invalid character in the DAT file. Less commonly something else crops up that I have to account for.

I'm on vacation at the moment and don't have access to my PC, so will look into this on my return.

**Comment by JbstormburstADV** [comment]
Bumping for awareness.

**Comment by unexpectedpanda** [comment]
Fixed in v2.01.6.
